File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        min_length = 200
        for st in strs:
            if len(st) < min_length:
                min_length = len(st)
        
        strs.sort()

        common_str = ""
        for i in range(min_length):
            if strs[0][i] != strs[-1][i]:
                return common_str
            common_str += strs[0][i]

        return common_str

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 1) {
            return strs[0];
        }

        int min_length = 200;
        for (String str: strs) {
            if (str.length() < min_length) {
                min_length = str.length();
            }
        }

        String common_str = "";
        for(int i = 0; i < min_length; i++) {
            char tmp_char = strs[0].charAt(i); 

            for (int j = 1; j < strs.length; j++) {
                if (strs[j].charAt(i) != tmp_char) {
                    return common_str;
                }
            }
            common_str += tmp_char;
        }

        return common_str;
    }
}